File: apps/admin/src/app/core/models/gym-membership.ts
export type MembershipStatus = 'pending' | 'active' | 'hidden' | 'inactive';

/**
 * The statuses an admin may assign. Mirrors `ManageableMembershipStatus` in the
 * contract: `pending` is owned by the join flow and the API rejects it with a
 * 400, so it is not a value any caller may send.
 */
export type SettableMembershipStatus = Exclude<MembershipStatus, 'pending'>;

export interface GymMembership {
  id: string;
  gymId: string;
  userId: string;
  status?: MembershipStatus;
  verifiedMember: boolean;
  gymRole?: string;
  isHome: boolean;
  visibleInRoster: boolean;
  joinMethod?: string;
  joinedAt: string;
  createdAt?: string;
  statusUpdatedAt?: string;
  statusUpdatedBy?: string;
}

File: apps/mobile/lib/features/membership/models/gym_membership.dart
class GymMembership {
  final String id;
  final String gymId;
  final String userId;
  final String status;
  final bool verifiedMember;
  final String gymRole;
  final bool isHome;
  final bool visibleInRoster;
  final String joinMethod;
  final String joinedAt;
  final String? createdAt;
  final String? statusUpdatedAt;
  final String? statusUpdatedBy;

  const GymMembership({
    required this.id,
    required this.gymId,
    required this.userId,
    required this.status,
    required this.verifiedMember,
    required this.gymRole,
    required this.isHome,
    required this.visibleInRoster,
    required this.joinMethod,
    required this.joinedAt,
    this.createdAt,
    this.statusUpdatedAt,
    this.statusUpdatedBy,
  });

  factory GymMembership.fromJson(Map<String, dynamic> json) => GymMembership(
    id: json['id'] as String,
    gymId: json['gymId'] as String,
    userId: json['userId'] as String,
    status: json['status'] as String,
    verifiedMember: json['verifiedMember'] as bool,
    gymRole: json['gymRole'] as String,
    isHome: json['isHome'] as bool,
    visibleInRoster: json['visibleInRoster'] as bool,
    joinMethod: json['joinMethod'] as String,
    joinedAt: json['joinedAt'] as String,
    createdAt: json['createdAt'] as String?,
    statusUpdatedAt: json['statusUpdatedAt'] as String?,
    statusUpdatedBy: json['statusUpdatedBy'] as String?,
  );
}
